Libraries/Text/Text/RCTTextRenderer.m

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#import "RCTTextRenderer.h"
+
+#import "RCTTextAttributes.h"
+
+@implementation RCTTextRenderer
+{
+  __weak CALayer *_layer;
+  NSTextStorage *_Nullable _textStorage;
+  CGRect _contentFrame;
+  CAShapeLayer *_highlightLayer;
+}
+
+- (instancetype)initWithLayer:(CALayer *)layer
+{
+  self = [super init];
+  if (self) {
+    _layer = layer;
+  }
+  return self;
+}
+
+- (void)setTextStorage:(NSTextStorage *)textStorage
+          contentFrame:(CGRect)contentFrame
+{
+  _textStorage = textStorage;
+  _contentFrame = contentFrame;
+}
+
+- (void)drawLayer:(CALayer *)layer
+        inContext:(CGContextRef)ctx;
+{
+  if (!_textStorage) {
+    return;
+  }
+
+  CGRect boundingBox = CGContextGetClipBoundingBox(ctx);
+  CGContextSaveGState(ctx);
+  UIGraphicsPushContext(ctx);
+
+  NSLayoutManager *layoutManager = _textStorage.layoutManagers.firstObject;
+  NSTextContainer *textContainer = layoutManager.textContainers.firstObject;
+
+  NSRange glyphRange =
+  [layoutManager glyphRangeForBoundingRect:boundingBox
+                           inTextContainer:textContainer];
+
+  [layoutManager drawBackgroundForGlyphRange:glyphRange atPoint:_contentFrame.origin];
+  [layoutManager drawGlyphsForGlyphRange:glyphRange atPoint:_contentFrame.origin];
+
+  __block UIBezierPath *highlightPath = nil;
```

Comment:
This part I'm not 100% sure about, what is this `_highlightLayer` for and how can I test it still renders properly?

Reply:
https://github.com/facebook/react-native/blob/master/Libraries/Text/RCTTextAttributes.h#L53